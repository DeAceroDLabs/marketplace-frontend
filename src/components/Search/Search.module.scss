@use "styles/colors";
@use "styles/typography";


.title {
    margin: 2%;
    display: flex;
    align-self: flex-end;
    .query {
        color: colors.$color-text-02;
        font-size: typography.$header-s-size;
        font-weight: typography.$header-s-weight;
        margin-right: 10px;
    }

}

.results {
    color: colors.$color-text-disabled;
    position: relative;
    top: 3px;

}