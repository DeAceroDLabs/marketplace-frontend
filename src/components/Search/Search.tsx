import React from "react";
import View from "components/common/View";
import styles from "./Search.module.scss";
import ProductsNotFound from "./ProductsNotFound";
import { useState, useEffect } from "react";
import { Product } from "config/api.types";
import { searchProducts } from "config/api";
import { useParams } from "react-router-dom";
import Card from "components/common/Card";
import GridContainer from "components/common/GridContainer";

const Search: React.FunctionComponent = () => {
  const [loading, setLoading] = useState(false);
  const { query } = useParams();
  const [products, setProducts] = useState<Product[]>([]);
  let productsCards = [];

  useEffect(() => {
    setProducts([]);
    setLoading(true);
    query &&
      searchProducts(query).then((response) => {
        setProducts(response.data);
        setLoading(false);
      });
  }, [query]);

  if (loading) {
    productsCards = Array.from(Array(12).keys()).map((item) => {
      return (
        <Card key={item} title={" "} size="medium" loading="loading-true" />
      );
    });
    return (
      <View>
        <div className={styles.info}>
          <div className={styles.query}>Cargando productos...</div>
        </div>
        <GridContainer>{productsCards}</GridContainer>
      </View>
    );
  }

  if (products.length > 0) {
    productsCards = products.map((product) => {
      return (
        <Card
          key={product.ClaProducto}
          title={product.NomProducto}
          imgSrc={product.CategoriaUrlImagen}
          size="medium"
          loading="loading-false"
        />
      );
    });
    return (
      <View>
        <div className={styles.info}>
          <div className={styles.query}>“{query}”</div>
          <div className={styles["results-number"]}>
            {products.length} resultados
          </div>
        </div>
        <GridContainer>{productsCards}</GridContainer>
      </View>
    );
  }

  return <ProductsNotFound />;
};

export default Search;