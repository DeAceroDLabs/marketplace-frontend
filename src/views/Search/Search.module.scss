@use "styles/colors";
@use "styles/typography";
.container {
    margin: auto;
    width: fit-content;
}

.info {
    display: flex;
    align-items: flex-end;
    .query {
        font-size: typography.$header-s-size;
        font-weight: typography.$header-s-weight;
        margin-right: 5px;
    }
    .results-number {
        font-size: typography.$text-body-s;
        color: colors.$color-text-disabled;
        position: relative;
        top: -2px;
    }
}