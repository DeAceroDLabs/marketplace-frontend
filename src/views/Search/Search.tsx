import React from "react";
import View from "components/common/View";
import styles from "./Search.module.scss";
import ProductsNotFound from "./ProductsNotFound";
import { useState, useEffect } from "react";
import { Product } from "config/api.types";
import { searchProducts } from "config/api";
import { useParams } from "react-router-dom";
import Card from "components/common/Card";
import LoadingCard from "components/common/LoadingCard";
import GridContainer from "components/common/GridContainer";
import Section from "components/common/Section";

const Search: React.FunctionComponent = () => {
  const [loading, setLoading] = useState(false);
  const { query } = useParams();
  const [products, setProducts] = useState<Product[]>([]);
  let productsCards;

  useEffect(() => {
    setProducts([]);
    setLoading(true);
    query &&
      searchProducts(query).then((response) => {
        setProducts(response.data);
        setLoading(false);
      });
  }, [query]);

  if (loading) {
    productsCards = <LoadingCard cards={12} variant="medium" />;
    return (
      <View>
        <Section>
          <div className={styles.container}>
            <div className={styles.info}>
              <div className={styles.query}>Cargando productos...</div>
            </div>
            <GridContainer>{productsCards}</GridContainer>
          </div>
        </Section>
      </View>
    );
  }

  if (products.length > 0) {
    productsCards = products.map((product) => {
      return (
        <Card
          key={product.ClaProducto}
          title={product.NomProducto}
          imgSrc={product.CategoriaUrlImagen}
          size="medium"
          loading={false}
        />
      );
    });

    return (
      <View>
        <Section>
          <div className={styles.container}>
            <div className={styles.info}>
              <div className={styles.query}>“{query}”</div>
              <div className={styles["results-number"]}>
                {products.length} resultados
              </div>
            </div>
            <GridContainer>{productsCards}</GridContainer>
          </div>
        </Section>
      </View>
    );
  }

  return <ProductsNotFound />;
};

export default Search;
