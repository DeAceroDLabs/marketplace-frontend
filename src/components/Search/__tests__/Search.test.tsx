import renderer from "react-test-renderer";
import Search from "../Search";

it("renders Search with no issue", () => {
  const component = renderer.create(<Search />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
