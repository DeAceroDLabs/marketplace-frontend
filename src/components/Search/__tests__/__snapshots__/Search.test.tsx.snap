// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Search with no issue 1`] = `
<div>
  <div
    className="container"
  >
    <img
      alt="Not found img"
      className="image"
      src="/not-found.jpg"
    />
    <div
      className="containerInfo"
    >
      <div
        className="title"
      >
        No encontramos un producto que coincida con tu búsqueda
      </div>
      <div
        className="info"
      >
        <div>
          Navega por las 
          <a
            className="a"
            href="/"
          >
            categorías
          </a>
           para encontrar un producto similar
        </div>
      </div>
    </div>
  </div>
</div>
`;
